mpl_interactions/generic.py

```diff
@@ -480,3 +482,221 @@ def _onselect(self, verts):
+         If False the function will try to detect if it is ok to use ipywidgets
+         If ipywidgets are not used the function will fall back on matplotlib widgets
+     play_buttons : bool or dict or list(str), optional
+        Whether to attach an ipywidgets.Play widget to any sliders that get created.
+        If a boolean it will apply to all kwargs, if a dictionary you choose which sliders you
+        want to attach play buttons too. If a list of strings use the names of the parameters that
+        you want to have sliders
+        left' or 'right'. Whether to position the play widget(s) to the left or right of the slider(s)
+    is_color_image : boolean
+        If True, will treat the last 3 dimensions as comprising a color images and will only set up sliders for the first arr.ndim - 3 dimensions.
+
+    returns
+    -------
+    fig : matplotlib figure
+    ax : matplotlib axis
+    controls : list of widgets
+    """
+
+    arr = np.asarray(np.squeeze(arr))
+    if arr.ndim < 3 + is_color_image:
+        raise ValueError(
+            f"arr must be at least {3+is_color_image}D but it is {arr.ndim}D. mpl_interactions.imshow for 2D images."
+        )
+
+    if is_color_image:
+        im_dims = 3
+    else:
+        im_dims = 2
+
+    params = {}
+    ipympl = notebook_backend()
+    fig, ax = gogogo_figure(ipympl, figsize, ax)
+    use_ipywidgets = ipympl or force_ipywidgets
+    slider_format_strings = create_slider_format_dict(slider_format_string, use_ipywidgets)
+
+    name_to_dim = {}
+    slices = [0 for i in range(arr.ndim - im_dims)]
+
+    names = None
+    axes = None
+    if "names" in kwargs:
+        names = kwargs.pop("names")
+
+    elif "axes" in kwargs:
+        axes = kwargs.pop("axes")
+
+    # Just pass in an array - no kwargs
+    for i in range(arr.ndim - im_dims):
+
+        slider_arr_passed = False
+        start, stop = None, None
+        name = f"axis{i}"
+        if name in kwargs:
+            if len(kwargs[name]) == 2:
+                start, stop = kwargs.pop(name)
+            else:
+                slider_arr_passed = True
+                slider_arr = kwargs.pop(name)
+
+        if names is not None and names[i] is not None:
+            name = names[i]
+
+        elif axes is not None and axes[i] is not None:
+            name = axes[i][0]
+
+            if len(axes[i]) == 2 and axes[i][1] is not None:
+                if len(axes[i][1]) == 2:
+                    start, stop = axes[i][1]
+                else:
+                    slider_arr_passed = True
+                    slider_arr = axes[i][1]
+
+        name_to_dim[name] = i
+
+        if slider_arr_passed:
+            kwargs[name] = slider_arr
+
+        else:
+
+            if start is None or stop is None:
+                kwargs[name] = np.arange(arr.shape[i])
+                slider_format_strings[name] = "{:d}"
+            else:
+                kwargs[name] = np.linspace(start, stop, arr.shape[i])
```

Comment:
this is fine, but you can also offload the linspacing to the `kwargs_to_widgets` function liek this
```suggestion
                kwargs[name] = (start, stop, arr.shape[i])
```